File: app/src/main/java/forpdateam/ru/forpda/api/search/models/SearchResult.java
package forpdateam.ru.forpda.api.search.models;

import java.util.ArrayList;
import java.util.List;

import forpdateam.ru.forpda.api.others.pagination.Pagination;

/**
 * Created by radiationx on 01.02.17.
 */

public class SearchResult {
    private List<SearchItem> items = new ArrayList<>();
    private SearchSettings settings;
    private Pagination pagination = new Pagination();
    private String html;

    public List<SearchItem> getItems() {
        return items;
    }

    public void addItem(SearchItem item) {
        items.add(item);
    }

    public SearchSettings getSettings() {
        return settings;
    }

    public void setSettings(SearchSettings settings) {
        this.settings = settings;
    }

    public Pagination getPagination() {
        return pagination;
    }

    public void setPagination(Pagination pagination) {
        this.pagination = pagination;
    }

    public String getHtml() {
        return html;
    }

    public void setHtml(String html) {
        this.html = html;
    }
}

File: app/src/main/java/forpdateam/ru/forpda/entity/remote/search/SearchResult.kt
package forpdateam.ru.forpda.entity.remote.search

import forpdateam.ru.forpda.entity.DeferredData
import forpdateam.ru.forpda.entity.remote.others.pagination.Pagination

/**
 * Created by radiationx on 01.02.17.
 */

data class SearchResult(
    val items: List<SearchItem>,
    val settings: SearchSettings,
    val pagination: Pagination,
    val html: DeferredData<String>?
) 
